experiments/functional/scale_jiva_replica/test.yml

```diff
@@ -86,6 +106,74 @@
 
+
+       - block:
+
+           - name: Getting the PV name
+             shell: kubectl get pvc {{ persistant_vol_name }} -n {{ namespace }} -o jsonpath='{.spec.volumeName}'
+             args:
+               executable: /bin/bash
+             register: pvname
+             changed_when: True
+          
+           - name: Waiting for replicas to start sync
+             shell: kubectl exec -it {{ maya_api_pod.stdout }} -n {{ operator_namespace }} -- mayactl volume describe --volname "{{ pvname.stdout }}" -n {{ namespace }} | grep 'WO' | wc -l
+             register: rep_sync_status
+             until: rep_sync_status.stdout|int != 0
+             delay: 5
+             retries: 30
+
+         when: lookup('env','COMPONET_FAILURE') == 'controller' or lookup('env','COMPONET_FAILURE') == 'replica'
+
+
+       - block:
+           
+           - name: Getting the Cotroller pod name
+             shell: kubectl get pod -n {{ namespace }} -l openebs.io/replica=jiva-replica -o jsonpath='{.items[0].metadata.name}'
+             register: ctrl_pod_name
+             
+           - name: Restarting controller
+             shell: kubectl delete pod {{ ctrl_pod_name.stdout }} -n {{ namespace }}
+             register: ctrl_delete
+             failed_when: "'deleted' not in ctrl_delete.stdout"
+
+           - name: Verifying successful deletion of controller
+             shell: kubectl get pod -n {{ namespace }} -l openebs.io/controller=jiva-controller -o jsonpath='{.items[0].metadata.name}'
+             register: ctrl_del_status
+             until: "'{{ ctrl_pod_name.stdout }}' not in ctrl_del_status.stdout"
+             delay: 10
+             retries: 100
+
+           - name: Waiting for new controller to be in running state
+             shell: kubectl get pod -n {{ namespace }} -l openebs.io/controller=jiva-controller -o jsonpath='{.items[0].status.phase}'
+             register: new_ctrl_state
+             until: "'Running' in new_ctrl_state.stdout"
+             delay: 10
+             retries: 100
+
+         when: lookup('env','COMPONET_FAILURE') == 'controller'
+
+
+       - block:
+          
+           - name: Getting the replica pod name
+             shell: kubectl get pod -n {{ namespace }} -l openebs.io/controller=jiva-controller -o jsonpath='{.items[0].metadata.name}'
+             register: rep_pod_name
+
+           - name: Restarting replica
+             shell: kubectl delete pod {{ rep_pod_name.stdout }} -n {{ namespace }}
+             register: replica_delete
+             failed_when: "'deleted' not in replica_delete.stdout"
+
+           - name: Verifying successful deletion of replica
+             shell: kubectl get pod -n {{ namespace }} -l openebs.io/replica=jiva-replica -o jsonpath='{.items[0].metadata.name}'
+             register: replica_del_status
+             until: "'{{ rep_pod_name.stdout }}' not in replica_del_status.stdout"
+             delay: 10
+             retries: 100
+
```

Comment:
check the replica count after failure

Reply:
@shashank855  check the replica status and wait until all the replicas are in Running state.